python/pyiceberg/utils/schema_conversion.py

```diff
@@ -64,11 +64,11 @@
 
 LOGICAL_FIELD_TYPE_MAPPING: Dict[Tuple[str, str], PrimitiveType] = {
     ("date", "int"): DateType(),
-    ("time-millis", "int"): TimeType(),
+    ("time-micros", "long"): TimeType(),
```

Comment:
We also want to be able to read `time-millis` right? We have readers for it in Java where we just multiple by 1_000 when we get the value.

Reply:
I'm fine either way. Eventually, we'll need to read files that were written for Hive and imported into an Iceberg table, so we are generally more permissive with reads. Since this is mostly about Iceberg metadata right now, we don't need to worry about it.